### PR TITLE
Allow patching wheels

### PR DESCRIPTION
**Pull request by mirandaconrado**
Sometimes we want to mokey patch python modules, either to adjust the behavior or to implement a fix
until it's updated in the package. This commit introduces a way to patch libraries by appending
their `__init__.py` file with new content from files.

### CODE REVIEW

File: python/whl.bzl [hunk 1]
```diff
@@ -40,6 +40,15 @@ def _extract_wheels(ctx, wheels):
     if result.return_code:
         fail("extract_wheels failed: %s (%s)" % (result.stdout, result.stderr))
 
+    patch_runtime = ctx.attr.patch_runtime
+    patch_runtime = [ctx.path(p) for p in patch_runtime]
+    name = ctx.attr.key
```
Comment (scele) [inline python/whl.bzl]: I don't think `<key>/__init__.py` is a valid logic for choosing the file to patch. First, there is not necessarily even a subdirectory in the wheel (e.g. six consists of just six.py), and second, the package name might not match the directory name (e.g. "tensorflow-gpu" is the package name, but top level directory is "tensorflow").
Also, sometimes we might want to patch some other file than `__init__.py`, so I wonder if we could make this more generic. How about instead having a `patch` attribute that would contain patch files, and we'd apply those patches? Then any file extracted from the wheel could be patched, or new ones added. Tensorflow repo uses similar idiom with their `patched_http_archive`. WDYT?

**Review by scele — CHANGES_REQUESTED** [review]

File: python/whl.bzl (hunk 1 above)
Comment (mirandaconrado) [reply python/whl.bzl]: Yep, makes more sense indeed.

Comment (mirandaconrado) [reply python/whl.bzl]: Fixed

File: python/whl.bzl [hunk 2]
```diff
@@ -172,6 +206,11 @@ extract_wheels = repository_rule(
             default = Label("//tools:piptool.par"),
             cfg = "host",
         ),
+        "_patcher": attr.label(
```
Comment (scele) [inline python/whl.bzl:209]: This is no longer needed?
Also, why is piptool.par being updated since you are not touching any python files?

**Comment by scele** [comment]
I made the small fixes and merged.

**Pull request closed without merging**